### PR TITLE
Implement `SET password_hash` for roles

### LINKED ISSUE #1492: Implement `CREATE ROLE SET password_hash`
This is needed for [`DESCRIBE ROLES`](https://github.com/edgedb/edgedb/issues/1491) to emit complete statements.

### PR DESCRIPTION
**Pull request by tailhook**
Fixes #1492

### CODE REVIEW

File: edb/schema/roles.py [hunk 1]
```diff
@@ -49,6 +49,13 @@ class Role(so.GlobalObject, so.InheritingObject,
         allow_ddl_set=True,
         inheritable=False)
 
+    password_hash = so.SchemaField(
```
Comment (elprans) [inline edb/schema/roles.py:52]: Why not just allow setting hash to the `password` field directly?

Comment (tailhook) [reply edb/schema/roles.py:52]: Do you mean detecting prefix in the password? I think it's weird. While it's unlikely that your password would start with `SCRUM-SHA-256$`, but still matching on that text sounds like a bad design. Or this isn't what you propose?

Comment (elprans) [reply edb/schema/roles.py:52]: OK, I wanted to see if you considered that option.  A separate DDL field is OK too.

File: edb/schema/roles.py [hunk 2]
```diff
@@ -72,9 +79,24 @@ def _process_role_body(
...
             cmd.set_attribute_value('password', salted_password)
 
+        password_hash = cmd.get_attribute_value('password_hash')
+        if password_hash is not None:
+            if not password_hash.startswith('SCRAM-SHA-256$'):
+                raise errors.InvalidValueError(
+                    'only SCRAM-SHA-256 hashes are supported',
+                    context=astnode.context,
+                )
+            cmd.set_attribute_value('password', password_hash)
```
Comment (elprans) [inline edb/schema/roles.py:98]: Please add a check that the hash is well-formed. (`scram.parse_verifier` should do the trick).

**Comment by Neustradamus** [comment]
Good job!

Linked to:
- https://github.com/scram-sasl/info/issues/1

**Pull request merged**